test/Microsoft.ML.Tests/Scenarios/Api/TestApi.cs

```diff
@@ -183,6 +183,95 @@ public void TrainAveragedPerceptronWithCache()
+            featureColumnWithMetadata.AddMetadata(kindFloatArray, valueFloatArray);
+            featureColumnWithMetadata.AddMetadata(kindVBuffer, valueVBuffer);
+
+            var mySchema = new SchemaDefinition { labelColumnWithMetadata, featureColumnWithMetadata };
+            var idv = mlContext.CreateDataView(data, mySchema);
+
+            Assert.True(idv.Schema[0].Metadata.Schema.Count == 2);
+            Assert.True(idv.Schema[0].Metadata.Schema[0].Name == kindFloat);
+            Assert.True(idv.Schema[0].Metadata.Schema[0].Type == coltypeFloat);
+            Assert.True(idv.Schema[0].Metadata.Schema[1].Name == kindString);
+            Assert.True(idv.Schema[0].Metadata.Schema[1].Type == TextType.Instance);
+
+            Assert.True(idv.Schema[1].Metadata.Schema.Count == 3);
+            Assert.True(idv.Schema[1].Metadata.Schema[0].Name == kindStringArray);
+            Assert.True(idv.Schema[1].Metadata.Schema[0].Type.IsVector && idv.Schema[1].Metadata.Schema[0].Type.ItemType.IsText);
+            var thrown = false;
+            try
+            {
+                var schema = idv.Schema[1].Metadata.Schema[kindFloat];
+            }
+            catch
+            {
+                thrown = true;
+            }
+            Assert.True(thrown);
+
+            float retrievedFloat = 0;
+            idv.Schema[0].Metadata.GetValue(kindFloat, ref retrievedFloat);
+            Assert.True(Math.Abs(retrievedFloat - valueFloat) < .000001);
+
+            ReadOnlyMemory<char> retrievedReadOnlyMemory = new ReadOnlyMemory<char>();
+            idv.Schema[0].Metadata.GetValue(kindString, ref retrievedReadOnlyMemory);
+            Assert.True(retrievedReadOnlyMemory.Span.SequenceEqual(valueString.AsMemory().Span));
+
+            VBuffer<ReadOnlyMemory<char>> retrievedReadOnlyMemoryVBuffer = new VBuffer<ReadOnlyMemory<char>>();
+            idv.Schema[1].Metadata.GetValue(kindStringArray, ref retrievedReadOnlyMemoryVBuffer);
+            Assert.True(retrievedReadOnlyMemoryVBuffer.DenseValues().Select((s, i) => s.ToString() == valueStringArray[i]).All(b => b));
+
+            VBuffer<float> retrievedFloatVBuffer = new VBuffer<float>(1, new float[] { 2 });
+            idv.Schema[1].Metadata.GetValue(kindFloatArray, ref retrievedFloatVBuffer);
+            VBuffer<float> valueFloatVBuffer = new VBuffer<float>(valueFloatArray.Length, valueFloatArray);
+            Assert.True(retrievedFloatVBuffer.Items().SequenceEqual(valueFloatVBuffer.Items()));
+
+            VBuffer<float> retrievedVBuffer = new VBuffer<float>();
+            idv.Schema[1].Metadata.GetValue(kindVBuffer, ref retrievedVBuffer);
+            Assert.True(retrievedVBuffer.Items().SequenceEqual(valueVBuffer.Items()));
+
+            try
+            {
+                idv.Schema[1].Metadata.GetValue(kindFloat, ref retrievedReadOnlyMemoryVBuffer);
+                Assert.True(false, "Throw an error if attribute is applied to a field that is not an IChannel.");
```

Comment:
>Throw an error if attribute is applied to a field that is not an IChannel. [](start = 36, length = 74)

WHAT?

Reply:
This was surprising to me as well  .  Its present in internal repo, so I kept it as is.

Deleted. Using Assert.Throws instead

---
In reply to: [240742454](https://github.com/dotnet/machinelearning/pull/1836#discussion_r240742454) [](ancestors = 240742454)

Reply:
My comment was about content assert :)

---
In reply to: [240745243](https://github.com/dotnet/machinelearning/pull/1836#discussion_r240745243) [](ancestors = 240745243,240742454)